pkg/abstractions/common/instance.go

```diff
@@ -287,3 +298,72 @@ func (i *AutoscaledInstance) State() (*AutoscaledInstanceState, error) {
 	state.FailedContainers = failedContainers
 	return &state, nil
 }
+
+func (i *AutoscaledInstance) HandleDeploymentDegraded(stubId string, containers []string) {
```

Comment:
It feels weird / redundant that we have an additional state, and an event. I would either remove the redis key or remove the event